Tests/Service/Request/RequestFactoryTest.php

```diff
@@ -166,12 +166,13 @@ public function testBuildFlow()
     public function testBuildFlowWithQueryParams()
     {
         $baseUrl = 'baseUrl';
-        $routeString = '/routeString?first-param={firstParam}&second-param=ignored value';
+        $routeString = '/routeString?first-param={firstParam}&second-param=ignored value&arr-param[key]={arrParam}';
```

Comment:
There's no test case for dot-notation, are you sure it's gonna work with `dot.separated[0].arrays` as well?

Reply:
Test added, and pass.